components/eam/src/physics/crm/vertical_diffusion.F90

```diff
@@ -132,1298 +106,765 @@ module vertical_diffusion
+                     history_budget_histfile_num_out = history_budget_histfile_num, &
+                     history_waccm_out = history_waccm)
+
+  if (history_amwg) then
+    call add_default(  vdiffnam(1), 1, ' ' )
+    call add_default( 'DTV'       , 1, ' ' )  
+    call add_default( 'PBLH'      , 1, ' ' )
+  endif
+  if( history_budget ) then
+    call add_default( vdiffnam(ixcldliq), history_budget_histfile_num, ' ' )
+    call add_default( vdiffnam(ixcldice), history_budget_histfile_num, ' ' )
+    if( history_budget_histfile_num > 1 ) then
+       call add_default(  vdiffnam(1), history_budget_histfile_num, ' ' )
+       call add_default( 'DTV'       , history_budget_histfile_num, ' ' )
     end if
+  end if
+
+  if ( history_waccm ) then
+    call add_default( 'DUV'     , 1, ' ' )
+    call add_default( 'DVV'     , 1, ' ' )
+  end if
+  !-----------------------------------------------------------------------------
+  !-----------------------------------------------------------------------------
+
+  qrl_idx   = pbuf_get_index('QRL')
+  wsedl_idx = pbuf_get_index('WSEDL')
+
+  ! Initialization of some pbuf fields
+  if (is_first_step()) then
+    ! Initialization of pbuf fields tke, kvh, kvm are done in phys_inidat
+    call pbuf_set_field(pbuf2d, turbtype_idx, 0    )
+    call pbuf_set_field(pbuf2d, smaw_idx,     0._r8)
+    call pbuf_set_field(pbuf2d, tauresx_idx,  0._r8)
+    call pbuf_set_field(pbuf2d, tauresy_idx,  0._r8)
+  end if
+
+end subroutine vertical_diffusion_init
+
+!===============================================================================
+!===============================================================================
+
+subroutine vertical_diffusion_ts_init( pbuf2d, state )
+  !-----------------------------------------------------------------------------
+  ! Timestep dependent setting,                                   
+  ! At present only invokes upper bc code for molecular diffusion 
+  !-----------------------------------------------------------------------------
+  use molec_diff    , only : init_timestep_molec_diff
+  use physics_types , only : physics_state
+  use ppgrid        , only : begchunk, endchunk
+  use physics_buffer, only : physics_buffer_desc
+  !-----------------------------------------------------------------------------
+  type(physics_state), intent(in) :: state(begchunk:endchunk)                 
+  type(physics_buffer_desc), pointer :: pbuf2d(:,:)
+  !-----------------------------------------------------------------------------
+  if (do_molec_diff) call init_timestep_molec_diff(pbuf2d, state )
+
+end subroutine vertical_diffusion_ts_init
+
+!===============================================================================
+!===============================================================================
+
+subroutine vertical_diffusion_tend( ztodt    , state    ,                  &
+                                        taux     , tauy     , shflx    , cflx, &
```

Comment:
now that you are making cosmetic changes, maybe you could align argument list here?

Reply:
fixed.